packages/database/.babelrc.js

```diff
@@ -0,0 +1,38 @@
+const env = {
+  test: {
+    plugins: ['istanbul'],
+  },
+};
+
+const includedDateOffset = 90 * 24 * 60 * 60 * 1000; // include migrations up to 90 days old
+const includedMigrationsDate = new Date().setTime(Date.now() - includedDateOffset);
+const checkMigrationWithinIncludedRange = function (migrationName) {
+  const yearPart = migrationName.substring(0, 4);
+  const monthPart = migrationName.substring(4, 6);
+  const dayPart = migrationName.substring(6, 8);
+  const migrationDate = new Date(`${yearPart}-${monthPart}-${dayPart}`);
+  return migrationDate < includedMigrationsDate;
+};
+
+const ignore = [
+  'src/tests/**',
+  function (filepath) {
+    const filepathComponents = filepath.split('/');
+    const directory =
+      filepathComponents.length > 2 ? filepathComponents[filepathComponents.length - 2] : null;
+    if (directory === 'migrations' || directory === 'migrationData') {
+      const filename = filepathComponents[filepathComponents.length - 1];
+      return checkMigrationWithinIncludedRange(filename);
+    }
+    const parentDirectory =
+      filepathComponents.length > 3 ? filepathComponents[filepathComponents.length - 3] : null;
+    if (parentDirectory === 'migrationData') {
+      // Some migration data is broken into separate subfiles within a folder named with the
+      // same name as the migration
+      return checkMigrationWithinIncludedRange(directory);
+    }
```

Comment:
Optional suggestion: using a stack can make the logic a bit simpler, since you can deconstruct the components in one go without checking for their length:

```js
const components = filepath.split('/');
const fileName = components.pop(); // will be always on top
const directory = components.pop(); // may be `undefined`
const parentDirectory = components.pop(); // may be `undefined`
```

Reply:
True, that's nice!

Reply:
Ah, but there's no `/` in the string...I remember getting to this point when I looked at it initially

Reply:
I think that's still fine?

```js
'fileName'.split('/') // ['test']
'/fileName'.split('/') // ['', 'fileName']

'directory/fileName'.split('/') // ['directory', 'fileName']
'/directory/fileName'.split('/') // ['', 'directory', 'fileName']
```

Unless if you mean something else

Reply:
Oh yes this is a different spot! I was looking at the date string. Shouldn't try to multi-task while I'm in a call. I'll come back to this at the end of my meetings for the day